lib/ansible/modules/uri.py

```diff
@@ -467,8 +467,9 @@ def write_file(module, dest, content, resp):
     """
     Create temp file and write content to dest file only if content changed
     """
+    fd, tmpsrc = tempfile.mkstemp(dir=module.tmpdir)
```

Comment:
I'm not positive about this one.  Before the `try/except` would catch errors making a tempfile too, but now it will not.

I think we need to wrap this individually from the `os.fdopen`, move the existing `module.fail_json` up here, and then reword the `module.fail_json` that handles opening the fd.

Reply:
That may have been the intent, but the existing code does not. Once the exception handler is reached, it would raise another exception due to `tmpsrc` being undefined. It could be fixed simply by setting `tmpsrc = None` before the `try` block and then checking if it is `None` before calling `os.path.exists`.